Scripts/Core/EditorVR.Menus.cs

```diff
@@ -197,119 +109,282 @@ internal void UpdateMenuVisibilities()
 					m_ActiveDeviceData.Add(deviceData);
 				});
 
-				// Reconcile conflicts because menus on the same device can visually overlay each other
-				for (int i = 0; i < m_ActiveDeviceData.Count; i++)
+				var directSelection = evr.GetNestedModule<DirectSelection>();
+
+				foreach (var deviceData in m_ActiveDeviceData)
 				{
-					var deviceData = m_ActiveDeviceData[i];
 					var alternateMenu = deviceData.alternateMenu;
 					var mainMenu = deviceData.mainMenu;
 					var customMenu = deviceData.customMenu;
 					var menuHideFlags = deviceData.menuHideFlags;
 
-					// Move alternate menu to another device if it conflicts with main or custom menu
-					if (alternateMenu != null && (menuHideFlags[mainMenu] == 0 || (customMenu != null && menuHideFlags[customMenu] == 0)) && menuHideFlags[alternateMenu] == 0)
+					var mainMenuVisible = mainMenu != null && (menuHideFlags[mainMenu] & MenuHideFlags.Hidden) == 0;
```

Comment:
Oof. Yeah, it's hard to know where to draw the line. Anything I could think to comment was a "what, not a why," so I opted to not write it.  I'll take another look and try to add comments or improve clarity.